src/main/MQ2Commands.cpp

```diff
@@ -4958,151 +4959,126 @@ void SetLootFilterCmd(SPAWNINFO* pChar, char* szLine)
 		// No arguments, display usage
 		if (!szLine[0])
 		{
-			WriteChatf("Usage: /setlootfilter (option) \"itemID^itemIconID^itemName\"");
+			WriteChatf("Usage: /setlootfilter (AN|AG|NV) [AR] itemID itemIconID \"itemName\"");
+			WriteChatf("Usage: /setlootfilter remove itemID");
 			WriteChatf("Options:");
 			WriteChatf("       AN - Always Need");
 			WriteChatf("       AG - Always Greed");
 			WriteChatf("       NV - Never");
 			WriteChatf("       AR - Always Roll");
-			WriteChatf("       ANAR - Always Need and Always Roll");
-			WriteChatf("       AGAR - Always Need and Always Greed");
-			WriteChatf("       NVAR - Never and Always Roll");
-			WriteChatf("Example: /setlootfilter ANAR \"13072^793^Rat Ears\"");
+			WriteChatf("Example: /setlootfilter AN AR 13072 793 \"Rat Ears\"");
 			WriteChatf("       Sets AdvLoot filter for \"Rat Ears\" to Always Need and Always Roll");
+			WriteChatf("Example: /setlootfilter remove 13072");
+			WriteChatf("       Removes AdvLoot filter for \"Rat Ears\"");
 			return;
 		}
 
-		// Grab user's selected filter type
-		char filterTypeArg[MAX_STRING] = { 0 };
-		GetArg(filterTypeArg, szLine, 1);
+		const char* szRest = szLine;
+		char szArg[MAX_STRING] = { 0 };
 
-		// Blank filter types, set by user below
+		// Grab Filter Type or Remove
+		GetArg(szArg, szRest, 1);
+		szRest = GetNextArg(szRest, 1);
+
+		// Check if we are doing a remove
+		if (ci_equals(szArg, "remove"))
+		{
+			// Grab Item ID
+			GetArg(szArg, szRest, 1);
+			szRest = GetNextArg(szRest, 1);
+
+			// Grab/Check Int of Item ID
+			int itemID = GetIntFromString(szArg, -1);
+			if (itemID == -1)
+			{
+				WriteChatf("Unable to remove AdvLoot Filter, Item ID Invalid.");
```

Comment:
minor nitpick here, but it's usually better to include the text that it tried to parse in the error message

Reply:
All the error messages should now print out what the argument was its complaining about.  Also missed the filter type argument having a check for it missing, which now has a message as well.